File: boot-pinyougou-sellergoods-interface/src/main/java/com/xt/pinyougou/service/GoodsService.java
package com.xt.pinyougou.service;

import com.baomidou.mybatisplus.core.metadata.IPage;
import com.baomidou.mybatisplus.extension.service.IService;
import com.xt.pinyougou.pojo.Goods;
import com.xt.pinyougou.pojogroup.GoodsGroup;

import java.util.List;

/**
 * <p>
 *  服务类
 * </p>
 *
 * @author xt
 * @since 2019-11-27
 */
public interface GoodsService extends IService<Goods> {

    void add(GoodsGroup goodsGroup);

    void update(GoodsGroup goodsGroup);

    IPage<Goods> selectPage(Integer currentPage, Integer pageNum, Goods goods);

    GoodsGroup findOne(Long id);

    void updateStatus(Long[] ids, String status);

    void deleteBatch(List<Long> asList);
}

File: boot-pinyougou-manager-web/src/main/resources/static/service/goodsService.js
// 商品服务
app.service('goodsService', function ($http) {

    // 保存
    this.add = function (entity) {
        return $http.post('/goods', entity);
    };

    // 更新
    this.update = function (entity) {
        return $http({
            method: 'PUT',
            url: '/goods',
            data: entity
        });
    };

    // 分页查询
    this.search = function(currentPage, pageNum, searchEntity) {
        return $http.post('/goods/page?currentPage=' + currentPage + '&pageNum=' + pageNum, searchEntity);
    };

    // 查询实体
    this.findOne = function (id) {
        return $http.get('/goods/' + id);
    };

    // 更新状态
    this.updateStatus = function (ids, status) {
        return $http.get('/goods/updateStatus/' + ids + '/' + status);
    };

    // 批量删除
    this.deleteBatch = function (ids) {
        return $http({
            method: 'DELETE',
            url: '/goods/deleteBatch/' + ids
        });
    };
});